### ISSUE TITLE
Missing syntax highlighting after Nix installation

### ISSUE THREAD

**Issue by noghartt** [warpdotdev/Warp#3142]
### Describe the bug

After installing [Nix](https://nixos.org), enabled `zsh` and restart the terminal, the syntax highlighting is missing some binaries that already exists on my `$PATH`. In comparison, iTerm is displaying it as what it should appears. But, in Warp, it seems that doesn't exist.

### To reproduce

1. Install Warp;
2. Setup Nix;
3. Enable `zsh` via Nix;
4. Enable `zsh-syntax-highlighting` plugin;
5. Restart Warp terminal;

### Expected behavior

It should be able to highlight the right command and display prompt suggestion about it, as expected.

### Screenshots

Warp:

![image](https://github.com/warpdotdev/Warp/assets/47952043/4ffee87f-ed26-4a93-aa8f-8adad5685b82)

iTerm:

![image](https://github.com/warpdotdev/Warp/assets/47952043/efea4056-3aa6-457c-a59d-9c14e7628d64)

### Operating system

MacOS

### Operating system and version

macOS Ventura 13.4

### Shell Version

zsh 5.9 (aarch64-apple-darwin21.6.0)

### Current Warp version

v0.2023.05.23.08.05.stable_00

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Recent working Warp date

Today, before Nix setup.

### Does this block you from using Warp daily?

No

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b8107fdf-ba31-488d-b103-d271c89cac3e

None

**Comment by zachbai** [comment]
Hey @noghartt thanks for filing. A couple follow-up questions:

* Is this happening within a "Warpfied" subshell (docs.warp.dev/features/subshells)
* Is the value of $PATH different between Warp + iTerm2?

**Comment by noghartt** [comment]
Yo, @zachbai! In case, answering your questions:

> * Is this happening within a "Warpfied" subshell (docs.warp.dev/features/subshells)

Nop, I'm not running a subshell in these cases, it's a recently opened terminal window.

> * Is the value of $PATH different between Warp + iTerm2?

Nop, both are identical.

**Comment by zachbai** [comment]
Oof, interesting.

Is syntax highlighting working with any non-builtin executables in your PATH? 

Non-builtin meaning something like `git` or `npm`, but not `ls` or `cd`.

**Comment by noghartt** [comment]
> Oof, interesting.
> 
> Is syntax highlighting working with any non-builtin executables in your PATH?
> 
> Non-builtin meaning something like `git` or `npm`, but not `ls` or `cd`.

Yes, they're working with `npm`, `git` and other executables, just brew and some others that didn't work.

**Comment by zachbai** [comment]
Wow, even more interesting. Is there a pattern with the path for executables that don't work? e.g. the parent directory from output of `which <brew|other executable>`? 

Is it possible that these binaries all live in a system or root directory that Warp doesn't have permissions to access, while the others live somewhere under your $HOME directory?

**Comment by noghartt** [comment]
> Wow, even more interesting. Is there a pattern with the path for executables that don't work? e.g. the parent directory from output of `which <brew|other executable>`?
> 
> Is it possible that these binaries all live in a system or root directory that Warp doesn't have permissions to access, while the others live somewhere under your $HOME directory?

I have some root directories in the `$PATH` that has been introduced by Nix (`/nix` and others), but Warp, apparently, have permission to access it.

The executable from brew has being target to the `/opt` dir that Warp has access too.

**Comment by zachbai** [comment]
Got it. Would you mind running `builtin print -l -- ${(ok)commands}` in Warp and sharing the block here via the 'Create permalink' button?

![image](https://github.com/warpdotdev/Warp/assets/10970845/493f4474-d441-4299-bd74-905e2d00045b)

This is the command Warp runs internally to fetch the list of valid executables in the session and syntax highlight appropriately.

**Comment by noghartt** [comment]
> Got it. Would you mind running `builtin print -l -- ${(ok)commands}` in Warp and sharing the block here via the 'Create permalink' button?
> 
> ![image](https://user-images.githubusercontent.com/10970845/243764927-493f4474-d441-4299-bd74-905e2d00045b.png)
> 
> This is the command Warp runs internally to fetch the list of valid executables in the session and syntax highlight appropriately.

Of course not, here's the [link](https://app.warp.dev/block/pWp96oqiGDONd5KhRrj7m3).

(By the way, that's a really cool feature)

**Comment by noghartt** [comment]
> Thanks @noghartt!
> 
> I don't have a working theory on the issue at the moment. What are some of the other commands that fail to syntax highlight?

> What are some of the other commands that fail to syntax highlight?

What I see is that it's always commands from `brew`. `code`, `yarn`, `npm`, `pnpm`, etc, all of these commands that inject some executable from brew on `$PATH` doesn't work the highlighting.

Other behavior that I saw is that it's working the highlighting on the terminal inside VSCode too, it's just happening on Warp.
